angular/src/components/generator.component.ts

```diff
@@ -93,19 +87,25 @@ export class PasswordGeneratorComponent implements OnInit {
       ) {
         this.usernameOptions.subaddressEmail = await this.stateService.getEmail();
       }
-      if (!this.showWebsiteOption) {
+      if (this.usernameWebsite == null) {
         this.usernameOptions.subaddressType = this.usernameOptions.catchallType = "random";
-      }
-      if (this.usernameWebsite != null) {
+      } else {
         this.usernameOptions.website = this.usernameWebsite;
+        const websiteOption = { name: this.i18nService.t("websiteName"), value: "website-name" };
+        this.subaddressOptions.push(websiteOption);
+        this.catchallOptions.push(websiteOption);
       }
 
-      if (qParams.type === "username" || qParams.type === "password") {
-        this.type = qParams.type;
-      } else {
-        const generatorOptions = await this.stateService.getGeneratorOptions();
-        if (generatorOptions != null && generatorOptions.type != null) {
-          this.type = generatorOptions.type;
+      if (this.type !== "username" && this.type !== "password") {
+        if (qParams.type === "username" || qParams.type === "password") {
+          this.type = qParams.type;
+        } else {
+          const generatorOptions = await this.stateService.getGeneratorOptions();
+          if (generatorOptions != null && generatorOptions.type != null) {
+            this.type = generatorOptions.type;
+          } else {
+            this.type = "password";
```

Comment:
Suggest `this.type = generatorOptions?.type ?? "password"` to minimise all this `if` nesting